src/Cmi5.ts

```diff
@@ -99,117 +141,145 @@ export default class Cmi5 {
       });
   }
 
-  public complete(): AxiosPromise<string[]> {
+  public complete(options: SendStatementOptions = {}): AxiosPromise<string[]> {
     // 10.0 xAPI State Data Model - https://github.com/AICC/CMI-5_Spec_Current/blob/quartz/cmi5_spec.md#100-xapi-state-data-model
-    if (this.launchData.launchMode !== "Normal") return Promise.reject();
-    return this.sendCmi5DefinedStatement({
-      // 9.3.3 Completed - https://github.com/AICC/CMI-5_Spec_Current/blob/quartz/cmi5_spec.md#933-completed
-      verb: Cmi5DefinedVerbs.COMPLETED,
-      result: {
-        // 9.5.3 Completion - https://github.com/AICC/CMI-5_Spec_Current/blob/quartz/cmi5_spec.md#953-completion
-        completion: true,
-        // 9.5.4.1 Duration - https://github.com/AICC/CMI-5_Spec_Current/blob/quartz/cmi5_spec.md#completed-statement
-        duration: XAPI.calculateISO8601Duration(
-          this.initialisedDate,
-          new Date()
-        ),
-      },
-      context: {
-        contextActivities: {
-          category: [
-            // 9.6.2.2 moveOn Category Activity - https://github.com/AICC/CMI-5_Spec_Current/blob/quartz/cmi5_spec.md#9622-moveon-category-activity
-            Cmi5ContextActivity.MOVE_ON,
-          ],
+    if (this.launchData.launchMode !== "Normal")
+      return Promise.reject(
+        new Error("Can only send COMPLETED when launchMode is 'Normal'")
+      );
+    return this.sendCmi5DefinedStatement(
+      {
+        // 9.3.3 Completed - https://github.com/AICC/CMI-5_Spec_Current/blob/quartz/cmi5_spec.md#933-completed
+        verb: Cmi5DefinedVerbs.COMPLETED,
+        result: {
+          // 9.5.3 Completion - https://github.com/AICC/CMI-5_Spec_Current/blob/quartz/cmi5_spec.md#953-completion
+          completion: true,
+          // 9.5.4.1 Duration - https://github.com/AICC/CMI-5_Spec_Current/blob/quartz/cmi5_spec.md#completed-statement
+          duration: XAPI.calculateISO8601Duration(
+            this.initialisedDate,
+            new Date()
+          ),
+        },
+        context: {
+          contextActivities: {
+            category: [
+              // 9.6.2.2 moveOn Category Activity - https://github.com/AICC/CMI-5_Spec_Current/blob/quartz/cmi5_spec.md#9622-moveon-category-activity
+              Cmi5ContextActivity.MOVE_ON,
+            ],
+          },
         },
       },
-    });
+      options
+    );
   }
 
   public pass(
-    score?: ResultScore,
-    objective?: ObjectiveActivity
+    score?: ResultScore | number,
+    objectiveOrOptions?: ObjectiveActivity | SendStatementOptions
```

Comment:
Why is this like this? Should they not be separate parameters?

Reply:
number 1 reason is backwards compatibility, and I'm assuming from the existing signature that you consider adding an objective activity common enough to make it a default param.

The thing I was trying to avoid was having a positional arg that callers would need to set null in order to use options, e.g.

```typescript
cmi.pass(0.9, null, { transform: (s) => ...}
```

...on the other hand, you want to make it easy to do maybe a transform *and* add the objective activity, so what do you think of just making `objectiveActivity` another property in options (last commit)?

Reply:
Currently, backwards compatibility is not top of the priorities list. As its still in pre-release, I want to make sure we get the structure of the signatures right before announcing a v1.0.0. I am possibly thinking that the first param should always be an object... I will take a look through the other methods and see if I can think of something that is future proof and user friendly.

Reply:
As I haven't had enough experience in using cmi5 in real life applications, I'm not certain how often one would need to add an objective activity. Everything that has been implemented has been based off of requirements in the cmi5 specification, and not from experience using cmi5 in the wild.

I suppose if both the objective activity and a custom transform is required, the objective activity could be added in the transformation?

Reply:
Yes, at the limit, can do anything/everything with the transform, but as a convenience, I also put `objectiveActivity` as another property of options. In future could put other common things that just get copied somewhere in the statement as properties in the options object e.g. `resultExtensions` and `contextExtensions`